halibot/halconfigurer.py

```diff
@@ -89,21 +89,26 @@ def validate_config(self, config, fill_default=True):
 		for key, validator in self.options.items():
 			tmp = config.get(key)
 
+			# Unsure on whether the item this one depends on NOT being present in config is warranted to make this required
+			required = not validator.depends or validator.depends not in config or config.get(validator.depends)
+
 			# Ensure the key exists, and if we aren't taking defaults, ensure we report
-			if not tmp and not fill_default:
+			if not tmp and not required:
+				break
```

Comment:
shouldn't this be a `continue` instead? if we break we don't validate the rest of the keys